libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -341,8 +341,11 @@ func (raw *cgroupData) join(subsystem string) (string, error) {
 	if err := os.MkdirAll(path, 0755); err != nil {
 		return "", err
 	}
-	if err := writeFile(path, CgroupProcesses, strconv.Itoa(raw.pid)); err != nil {
-		return "", err
+	// Dont attach any pid to the cgroup if -1 is specified as a pid
+	if raw.pid != -1 {
```

Comment:
i worry the -1 is kind of undocumented unless we have it that all writeFile calls to cgroup.procs share a common function call.  a lot of this code is not unit tested, so would prefer that we at least consolidate the -1 checking into a common function like `writeCgroupProc(pid int)` and just keep the -1 check local to there..